src/main/java/acme/entities/patronage/Patronage.java

```diff
@@ -0,0 +1,77 @@
+package acme.entities.patronage;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Positive;
+
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.URL;
+
+import acme.framework.datatypes.Money;
+import acme.framework.entities.AbstractEntity;
+import acme.roles.Inventor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+public class Patronage extends AbstractEntity{
+
+	// Serialisation identifier
+	
+	protected static final long serialVersionUID = 1L;
+	
+	// Attributes
+	
+	protected Status status;
+	
+	@Column(unique = true)
+	@Pattern(regexp="^[A-Z]{3}-[0-9]{3}(-[A-Z])?$")
+	private String code;
+	
+	@NotBlank
+	@Length(max=255)
+	private String legislation;
+	
+	@NotNull
+	@Positive //not sure if this is enough
```

Comment:
@Positive is not functional for complex types like Money, you need a complex constraint

Reply:
ok then it is left without the constraint for now, since it is not needed until unit 4